api/spec/scanner-adapter-openapi-v1.2.yaml

```diff
@@ -0,0 +1,546 @@
+            'sbom' represents generating the corresponding sbom for the artifact which be scanned. In order to the backward and forward compatible,
+            the field is optional, we think it's a original 'vulnerability' scan if no such field.
+        consumes_mime_types:
+          type: array
+          items:
+            type: string
+          description: |
+            The set of MIME types of the artifacts supported by the scanner to produce the reports specified in the "produces_mime_types". A given
+            mime type should only be present in one capability item.
+          example:
+            - "application/vnd.oci.image.manifest.v1+json"
+            - "application/vnd.docker.distribution.manifest.v2+json"
+        produces_mime_types:
+          type: array
+          items:
+            type: string
+          description: |
+            The set of MIME types of reports generated by the scanner for the consumes_mime_types of the same capability record.
+          example:
+            - "application/vnd.scanner.adapter.vuln.report.harbor+json; version=1.0"
+    ScanRequest:
+      required:
+        - registry
+        - artifact
+      type: object
+      properties:
+        registry:
+          $ref: '#/components/schemas/Registry'
+        artifact:
+          $ref: '#/components/schemas/Artifact'
+        enabled_capabilities:
+          type: array
+          description: Enable which capabilities supported by scanner, for backward compatibility, without this field scanner can be considered to enable all capabilities by default.
+          items:
+            type: object
+            properties:
+              type:
+                type: string
+                enum:
+                - vulnerability
+                - sbom
+                description: The type of the scan capability.
+                example: sbom
+              produces_mime_types:
+                type: array
+                items:
+                  type: string
+                description: |
+                  The set of MIME types of reports generated by the scanner for the consumes_mime_types of the same capability record, it is a subset or fullset of the
+                  produces_mime_types of the capability returned by the metadata API, used for client to fine grained control of the expected report type. It's a optional
+                  field, only applied when client needs to customize it, otherwise the scanner can think it's a fullset as before behavior if without this field.
+                example:
+                  - "application/vnd.security.sbom.report+json; version=1.0"
+              parameters:
+                type: object
+                additionalProperties: true
+                description: The additional parameters for the scan request.
+                example: '{"sbom_accept_media_type": "application/spdx+json"}'
```

Comment:
If scanners must follow the value specified in `sbom_accept_media_type`, shouldn't we explicitly define the field in the OpenAPI spec? It is a hint, and then can the scanner ignore that?

Reply:
The parameters is a common property which used for passing some domain related configs to the scanner, for sbom it's the `sbom_accept_media_type `, for other type there may be other properties, so define an explicitly key in a universal may not suitable, but harbor will do the validation to check whether the media type supported when receive the reports.

Reply:
Yes, a universal object doesn't make sense, but it is currently an example. It is hard for scanners to interpret the parameters. What if defining several parameters with `oneOf` as below? Then, we can explicitly define `sbom_accept_media_type` in `SbomParameters`.

```
          parameters:
            oneOf:
              - $ref: '#/components/schemas/VulnerabilityParameters'
              - $ref: '#/components/schemas/SbomParameters'
            discriminator:
              propertyName: type
              mapping:
                vulnerability: '#/components/schemas/VulnerabilityParameters'
                sbom: '#/components/schemas/SbomParameters'
```

Reply:
Good idea.